src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -1288,6 +1292,80 @@ public static Builder builder(MetaData metaData) {
         return new Builder(metaData);
     }
 
+    /** All known byte-sized cluster settings. */
+    public static final Set<String> CLUSTER_BYTES_SIZE_SETTINGS = ImmutableSet.of(
+        IndicesStore.INDICES_STORE_THROTTLE_MAX_BYTES_PER_SEC,
+        RecoverySettings.INDICES_RECOVERY_FILE_CHUNK_SIZE,
+        RecoverySettings.INDICES_RECOVERY_TRANSLOG_SIZE,
+        RecoverySettings.INDICES_RECOVERY_MAX_BYTES_PER_SEC,
+        RecoverySettings.INDICES_RECOVERY_MAX_SIZE_PER_SEC);
+
+
+    /** All known time cluster settings. */
+    public static final Set<String> CLUSTER_TIME_SETTINGS = ImmutableSet.of(
+                                    IndicesTTLService.INDICES_TTL_INTERVAL,
+                                    RecoverySettings.INDICES_RECOVERY_RETRY_DELAY_STATE_SYNC,
+                                    RecoverySettings.INDICES_RECOVERY_RETRY_DELAY_NETWORK,
+                                    RecoverySettings.INDICES_RECOVERY_ACTIVITY_TIMEOUT,
+                                    RecoverySettings.INDICES_RECOVERY_INTERNAL_ACTION_TIMEOUT,
+                                    RecoverySettings.INDICES_RECOVERY_INTERNAL_LONG_ACTION_TIMEOUT,
+                                    DiskThresholdDecider.CLUSTER_ROUTING_ALLOCATION_REROUTE_INTERVAL,
+                                    InternalClusterInfoService.INTERNAL_CLUSTER_INFO_UPDATE_INTERVAL,
+                                    InternalClusterInfoService.INTERNAL_CLUSTER_INFO_TIMEOUT,
+                                    DiscoverySettings.PUBLISH_TIMEOUT,
+                                    InternalClusterService.SETTING_CLUSTER_SERVICE_SLOW_TASK_LOGGING_THRESHOLD);
+
+    /** As of 2.0 we require units for time and byte-sized settings.  This methods adds default units to any cluster settings that don't
+     *  specify a unit. */
+    public static MetaData addDefaultUnitsIfNeeded(ESLogger logger, MetaData metaData) {
+        Settings.Builder newPersistentSettings = null;
+        for(Map.Entry<String,String> ent : metaData.persistentSettings().getAsMap().entrySet()) {
+            String settingName = ent.getKey();
+            String settingValue = ent.getValue();
+            if (CLUSTER_BYTES_SIZE_SETTINGS.contains(settingName)) {
+                try {
+                    Double.parseDouble(settingValue);
+                } catch (NumberFormatException nfe) {
+                    continue;
+                }
+                // It's a naked number; add default unit (b for bytes):
+                logger.warn("byte-sized cluster setting [{}] with value [{}] is missing units; now adding default units (b)", settingName, settingValue);
+                if (newPersistentSettings == null) {
```

Comment:
It would be great for this to be more of a deprecation warning. More like: "you need to fix this for ES 3.0, we won't be lenient there".